mapping/react-leaflet.md

```diff
@@ -0,0 +1,434 @@
+
+    const layerFeature = layer?.feature?.properties
+                          ? layer.feature.properties 
+                          : null
+    
+    setWard(layerFeature)
+  }
+
+  function eventHandlersOnEachFeature(feature, layer) {
+    layer.on({
+      click: onWardClick
+    })
+  }
+
+  return (
+    <>
+      <BaseMap center={[41.8781, -87.6298]} zoom={10}>
+        {/* this will only show when wardBorders has a value */}
+        {wardBorders && <GeoJSON
+                          key='ward-layer'
+                          data={wardBorders}
+                          style={fill} 
+                          onEachFeature={eventHandlersOnEachFeature} />}
+      </BaseMap>
+      {ward && <p>Ward {ward.ward}'s shape_area = {ward.shape_area} and shape_leng = {ward.shape_leng}</p>}
+    </>
+  )
+}
+
+export default ChicagoWardMap
+```
+
+Nice!
+
+### Lift state up
+One thing I don't like about the current design of the `ChicagoWardMap`: what if you need to use the ward's information elsewhere in the UI, in a place that is outside of this component? You would need to "lift state up" and set the `ward` state in a parent component. This is a common pattern in React applications, and can be difficult to understand if you're not familiar with it. It's simply a JavaScript callback that passes a message from a child component to a parent component.
+
+In your `App.js`, import the `useState` hook from React, create a `ward` state object, and render it on the page like you did in `ChicagoWardMap`:
+```jsx
+import { useState } from 'react'
+import './App.css'
+import ChicagoWardMap from './maps/wards'
+
+function App() {
+  const [ward, setWard] = useState(null)
+
+  return (
+    <div className="App">
+      <div className='map-viewer'>
+        <ChicagoWardMap 
+          onSelectWard={setWard}
+        />
+        {ward && <p>Ward {ward.ward}'s shape_area = {ward.shape_area} and shape_leng = {ward.shape_leng}</p>}
+      </div>
+    </div>
+  );
+}
+
+export default App
+```
+
+This code does what `ChicagoWardMap` did, but it depends on the `onSelectWard` prop to do this. So, we need to add that prop to the `ChicagoWardMap` component. (TODO: this is where a diff would be helpful...) Where you declare the `ChicagoWardMap` function, pass in `onSelectWard` as a prop:
+```jsx
+function ChicagoWardMap({ onSelectWard }) {...}
+```
+
+Remove these two lines of code, since you replaced them in `App.js`:
+- `const [ward, setWard] = useState(null)`
+- `{ward && <p>Ward {ward.ward}'s shape_area = {ward.shape_area} and shape_leng = {ward.shape_leng}</p>}`
+
+And in the `onWardClick` function, replace the `setWard` function with `onChooseCounty`:
+```jsx
+function onCountyClick(e) {
+  const layer = e.target
+
+  const layerFeature = layer?.feature?.properties
+                        ? layer.feature.properties 
+                        : null
+
+  // send the information about the feature to the parent component
+  onChooseCounty(layerFeature)
+}
+```
```

Comment:
I think this would be easier to follow if describe the changes, then said "Here's what your updated wards.js should look like" with the full updated component.